src/azure-cli/azure/cli/command_modules/monitor/_params.py

```diff
@@ -353,4 +353,36 @@ def load_arguments(self, _):
                         "all monitor settings would be cloned instead of expanding its scope.")
         c.argument('monitor_types', options_list=['--types', '-t'], arg_type=get_enum_type(['metricsAlert']),
                    nargs='+', help='List of types of monitor settings which would be cloned.', default=['metricsAlert'])
+
+    # region Private Link Resources
+    for item in ['create', 'update', 'show', 'delete', 'list']:
+        with self.argument_context('monitor private-link-scope {}'.format(item)) as c:
+            c.argument('scope_name', options_list=['--name', '-n'], help='Name of the Azure Monitor Private Link Scope.')
+    with self.argument_context('monitor private-link-scope create') as c:
+        c.ignore('location')
+
+    with self.argument_context('monitor private-link-scope scoped-resource') as c:
+        c.argument('scope_name', help='Name of the Azure Monitor Private Link Scope.')
+        c.argument('resource_name', options_list=['--name', '-n'], help='Name of the assigned resource.')
+        c.argument('linked_resource_id', options_list=['--linked-resource'], help='Name or ID of the linked resource. It should be one of log analytics workspace or application insights component.')
```

Comment:
Where do you convert name to id as I didn't find the related code?

Reply:
I think we should just support id. I will change the help message.